oceanmodes/baroclinic.py

```diff
@@ -165,7 +168,240 @@ def _neutral_modes_from_N2_profile_raw(z, N2, f0, depth=None, **kwargs):
     j = np.argsort(w)[::-1]
     w = w[j]
     v = v[:,j]
+    
+    #########
+    # w = - 1/(Rd^2 * f0^2)
+    #########
+    Rd = (-w)**-0.5 / f0
+
+    return zf, Rd, v
+
+def instability_analysis_from_N2_profile(z, N2, f0, beta, Nx, Ny, dx, dy, ubar, vbar, etax, etay, sort='LI', num=4, depth=None, **kwargs):
+    """Calculate baroclinic neutral modes from a profile of 
+        buoyancy frequency truncated at the bottom.
+    """
+    nz_orig = len(z)
+    z, N2 = _maybe_truncate_above_topography(z, N2)
+    return _instability_analysis_from_N2_profile_raw(
+                                    z, N2, f0, beta, Nx, Ny, dx, dy, ubar, vbar, etax, etay, sort, num, depth=depth, **kwargs)
+
+def _instability_analysis_from_N2_profile_raw(z, N2, f0, beta, Nx, Ny, dx, dy, ubar, vbar, etax, etay, sort='LI', num=4, depth=None, **kwargs):
+    """Calculate baroclinic unstable modes from a profile of buoyancy frequency.
+    
+        Solves the eigenvalue problem
+
+            \left ( \frac{\partial}{\partial t} + U \cdot \nabla \right ) \left ( \nabla^2 + \frac{d}{dz} \frac{f_0^2}{N^2} \frac{d \psi}{d z} \right )
+            + u \cdot \nabla Q
+            = 0
+
+        With the boundary conditions
+
+            \left ( \frac{\partial}{\partial t} + U \cdot \nabla \right ) f_0 \frac{\partial \psi}{\partial z}
+            + u \cdot \nabla \left ( B + N^2 \eta \right )
+            = 0
+
+        at the top and bottom.
+    
+        Parameters
+        ----------
+        z : array_like
+            The depths at which N2 is given. Starts shallow, increases
+            positive downward.
+        N2 : array_like
+            The squared buoyancy frequency (units s^-2). Points below topography
+            should be masked or nan.
+        f0 : float
+            Coriolis parameter.
+        Q : array_like
+            The background QGPV (\beta y + \left ( \nabla^2 + \frac{d}{dz} \frac{f_0^2}{N^2} \frac{d \Psi}{d z} \right ))
+            and Psi is the streamfunction of the background velocity.
+        eta : array_like
+            The topographic change at the top and bottom.
+        B : array_like
+            The background buoyancy
+        depth : float, optional
+            Total water column depth. If missing will be inferred from z.
+        kwargs : optional
+            Additional parameters to pass to scipy.sparse.linalg.eigs for
+            eigenvalue computation
```

Comment:
There are lots of function arguments that are not documented in the docstring (nx, ny, dx, dy, etc.)